src/web/lib/getCountryCode.ts

```diff
@@ -1,76 +1,56 @@
-type LocalCountryCodeType = {
-	value: string;
-	expires: number;
-};
-
-const COUNTRY_CODE_KEY = 'gu.geolocation';
-const TEN_DAYS = 60 * 60 * 24 * 10;
+import { getLocale, storage } from '@guardian/libs';
+import type { CountryCode } from '@guardian/libs/dist/esm/types/countries';
+import { getCookie } from '@frontend/web/browser/cookie';
 
-function hasExpired(whenItExpires: number) {
-	return new Date().getTime() > whenItExpires;
-}
-
-export const setCountryCode = (countryCode: string): void => {
-	// What's this setTimeout business?
-	// localStorage calls are syncronous and we don't need to wait for this
-	// one so we use setTimeout to put this step on the end of the event queue
-	// for later, letting the thread continue
-	setTimeout(() => {
-		if (countryCode) {
-			try {
-				localStorage.setItem(
-					COUNTRY_CODE_KEY,
-					JSON.stringify({
-						value: countryCode,
-						expires: new Date().getTime() + TEN_DAYS,
-					}),
-				);
-			} catch (error) {
-				// We tried, it failed. Often local storage is not available and we
-				// need to live with that
-			}
-		}
-	});
-};
+const COUNTRY_CODE_KEY = 'GU_geo_country';
+const COUNTRY_CODE_KEY_OVERRIDE = 'gu.geo.override';
+/*
+	Memoizes the location
+ */
+let locale: CountryCode | null;
 
-export const getCountryCode = async (): Promise<string | void> => {
-	// Read local storage to see if we already have a value
-	let localCountryCode: LocalCountryCodeType | null;
-	try {
-		const item = localStorage.getItem(COUNTRY_CODE_KEY);
-		localCountryCode = item ? JSON.parse(item) : null;
-	} catch (error) {
-		localCountryCode = null;
+/*
+	Can be used to override country code and sets it in localStorage
+ */
+export const overrideCountryCode = (countryCode: CountryCode): void => {
+	if (countryCode) {
+		storage.local.set(COUNTRY_CODE_KEY_OVERRIDE, countryCode);
+		locale = countryCode;
 	}
+};
 
-	if (!localCountryCode || hasExpired(localCountryCode.expires)) {
-		const countryCode = await fetch(
-			'https://api.nextgen.guardianapps.co.uk/geolocation',
-		)
-			.then((response) => {
-				if (!response.ok) {
-					throw Error(
-						response.statusText ||
-							`getCountryCode | An api call returned HTTP status ${response.status}`,
-					);
-				}
-				return response;
-			})
-			.then((response) => response.json())
-			.then((json: { country: string }) => json.country)
-			.catch((error) => {
-				window.guardian.modules.sentry.reportError(
-					error,
-					'get-country-code',
-				);
-			});
+/*
+	This method can be used as a non async way of getting the country code
+	after getLocaleCode has been called.
+ */
+export const getCountryCode = (): string | null => {
+	return (
+		locale ??
+		storage.local.get(COUNTRY_CODE_KEY_OVERRIDE) ??
+		getCookie(COUNTRY_CODE_KEY)
+	);
+};
```

Comment:
possibly only for lotame, good observation. I decided to keep the same in regards to sync/async as it was before, but def async is the way to go and rely completely on libs/getLocale.

For reference, in frontend we only call async once on init and then keep using the sync one. So decided to leave the same and do the conversion to full async as a separate task - it would have been 2 many changes to review.

I will rename it the method name hopefully will be less confusing